dandiapi/api/views/asset.py

```diff
@@ -137,11 +138,35 @@ def download(self, *args, **kwargs):
 
         # Redirect to correct presigned URL
         storage = asset_blob.blob.storage
-        return HttpResponseRedirect(
-            storage.generate_presigned_download_url(
-                asset_blob.blob.name, os.path.basename(asset.path)
+
+        serializer = AssetDownloadQueryParameterSerializer(data=request.query_params)
+        serializer.is_valid(raise_exception=True)
+        content_disposition = serializer.validated_data['content_disposition']
+        content_type = asset.metadata.get('encodingFormat', 'application/octet-stream')
+        asset_basename = os.path.basename(asset.path)
+
+        if content_disposition == 'attachment':
+            return HttpResponseRedirect(
+                storage.generate_presigned_download_url(asset_blob.blob.name, asset_basename)
             )
-        )
+        elif content_disposition == 'inline':
+            url = storage.generate_presigned_inline_url(
+                asset_blob.blob.name,
+                asset_basename,
+                content_type,
+            )
+
+            if content_type == 'video/x-matroska':
+                return HttpResponse(
+                    f"""
+                    <video autoplay muted controls>
+                        <source src="{url}" type="video/mp4">
+                    </video>
+                """,
+                    content_type='text/html',
+                )
```

Comment:
> I think this particular aspect might better be gone into a dedicated "Play video" within this registry of "services"

The service registry is for external software that can work with a DANDI asset URL. This solution just leverages the `<video>` tag to play .mkv files in the browser. Why would we want an external service for this?

> since otherwise user would not be able to just delegate it "as is" to the browser

If you "delegate" a .mkv file to the browser, the browser falls back to downloading it. The user can do that with the existing download button in this case (and without this special casing, the "view" button would result in the browser trying to download the mkv file, which is not what we want).

> (AFAIK matroska and mp4 are two different "competing" container formats, so not even sure if would always work reliably here, and why not to match for video/mp4 as well etc)

The browser can natively play mp4 files, so there's no need to match against `video/mp4` -- the fall-through case of "anything besides matroska" handles this just fine. Which part of this are you concerned may not always work?

Reply:
Thanks for the explanation of mp4 vs matroshka handling -- it makes more sense now, that it is a special workaround/shim for `mkv` files while browser plays `mp4` natively when "viewed".  Makes more sense now and indeed probably no need for including into Services context menu.